src/capwap_loc_provider_http.erl

```diff
@@ -0,0 +1,63 @@
+%% Copyright 2017-2019 Travelping GmbH <info@travelping.com>
+%%
+%% This program is free software: you can redistribute it and/or modify
+%% it under the terms of the GNU Lesser General Public License as
+%% published by the Free Software Foundation, either version 3 of the
+%% License, or (at your option) any later version.
+%%
+%% This program is distributed in the hope that it will be useful,
+%% but WITHOUT ANY WARRANTY; without even the implied warranty of
+%% MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+%% GNU Lesser General Public License for more details.
+%%
+%% You should have received a copy of the GNU Lesser General Public License
+%% along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+-module(capwap_loc_provider_http).
+
+-compile({parse_transform, cut}).
+
+-behaviour(capwap_loc_provider).
+
+%% AAA API
+-export([config_fun/1]).
+
+-include_lib("kernel/include/logger.hrl").
+
+-define(LAT_KEY, <<"latitude">>).
+-define(LONG_KEY, <<"longitude">>).
+
+config_fun(#{uri := Uri, timeout := Timeout}) ->
+  ParsedUri = uri_string:parse(Uri),
+  fun(Name) -> resolve_loc_http(ParsedUri, Timeout, Name) end.
+
+resolve_loc_http(Uri, Timeout, Name) ->
+    ?LOG(debug, "HTTP provider towards: ~p", [Uri]),
+    #{path := UriPath} = Uri,
+    JsonHeader = {<<"Content-Type">>, <<"application/json">>},
+    UriPathBin = unicode:characters_to_binary(UriPath),
+    ?LOG(debug, "UriPathBin: ~p", [UriPathBin]),
+    CompletePath = <<UriPathBin/binary, "/location/", Name/binary>>,
+    CompleteUri = Uri#{path => CompletePath},
```

Comment:
Hmmmmmm, I'm a bit worried about unicode conversion. A plain iolist_to_binary might mess things up. But maybe it's more readable using it instead of direct binary composition, yes, after using unicode:chars_to_bin.

We can include the Uri patch check in the config parameter checks, maybe? I mean, we should assume that configured data should be correct at this stage, right?

Reply:
URI can not contain unicode characters, see https://www.rfc-editor.org/rfc/rfc3986#section-3.3

Reply:
Suggestion taken after some URI RFC checks.